lua/rocks-config/init.lua

```diff
@@ -33,18 +33,30 @@ local function create_plugin_heuristics(name)
     })
 end
 
----Emulates Lua's require mechanism behaviour. Lua's `require` function
----returns `true` if the module returns nothing (`nil`), so we do the same.
----@param searcher function The module name
----@param mod_name string The module name
----@return any loaded
-local function try_load_like_require(searcher, mod_name)
-    local loader = searcher(mod_name)
+---Tries to get a loader function for a given module.
+---Returns nil if the module is not found.
+---@param mod_name string The module name to search for
+---@return function | nil
+local function try_get_loader_for_module(mod_name)
+    -- Loaders that search `package.preload` and `package.path`.
+    -- We don't need to search the cpath or neovim's runtimepath,
+    -- as the nvim `lua` directory is added to the `package.path`.
+    for _, searcher in ipairs(package.loaders) do
+        local loader = searcher(mod_name)
 
-    if type(loader) ~= "function" then
-        return nil
+        if type(loader) == "function" then
+            return loader
+        end
     end
 
+    return nil
+end
+
+---Emulates Lua's require mechanism behaviour. Lua's `require` function
+---returns `true` if the module returns nothing (`nil`), so we do the same.
+---@param loader function The loader function
+---@return any loaded
```

Comment:
issue: `any` is an EmmyLua/typescript type.
We use luaCATS.

```suggestion
---@return unknown loaded
```

Reply:
Sorry, I'm not familiar with Lua type-checking. I'm using LuaLS and as long as it isn't yelling at me I assume everything is fine :sweat_smile: Fixed.

Reply:
It is a lot more lenient than I am comfortable with :sweat_smile: